HW0/P4/P4.py

```diff
@@ -2,6 +2,18 @@
 import matplotlib.pyplot as plt
 from mpl_toolkits.mplot3d import Axes3D
 
+def predictS(s, A, a):
+    return np.dot(A, s) + a
+
+def predictSig(Sigma, A, B):
+    return np.linalg.inv(np.dot(np.dot(A, Sigma), np.transpose(A)) + np.dot(B, np.transpose(B)))
+
+def updateSig(Sigma_tilde, C):
+    return np.linalg.inv(Sigma_tilde + np.dot(np.transpose(C), C))
+
+def updateS(s_tilde, Sigma_tilde, Sigma, C, m):
+    return np.dot(Sigma, np.dot(Sigma_tilde, s_tilde) + np.dot(np.transpose(C), m))
+
 
```

Comment:
These equations all look clean and well written. It's possible to do C.transpose() or even C.T() instead of np.transpose, as someone noted on my own pull request. The only suggestion I have is to describe the equations or parameters for what each equation calculates and updates, as the variable names can get cryptic otherwise. But that's minor, great code!